web_timeline/static/src/js/timeline_renderer.js

```diff
@@ -0,0 +1,536 @@
+odoo.define('web_timeline.TimelineRenderer', function (require) {
+    "use strict";
+
+    var AbstractRenderer = require('web.AbstractRenderer');
+    var core = require('web.core');
+    var time = require('web.time');
+    var utils = require('web.utils');
+    var session = require('web.session');
+    var QWeb = require('web.QWeb');
+    var field_utils = require('web.field_utils');
+    var TimelineCanvas = require('web_timeline.TimelineCanvas');
+
+
+    var _t = core._t;
+
+    var TimelineRenderer = AbstractRenderer.extend({
+        template: "TimelineView",
+
+        events: _.extend({}, AbstractRenderer.prototype.events, {
+            'click .oe_timeline_button_today': '_onTodayClicked',
+            'click .oe_timeline_button_scale_day': '_onScaleDayClicked',
+            'click .oe_timeline_button_scale_week': '_onScaleWeekClicked',
+            'click .oe_timeline_button_scale_month': '_onScaleMonthClicked',
+            'click .oe_timeline_button_scale_year': '_onScaleYearClicked',
+        }),
+
+        init: function (parent, state, params) {
+            this._super.apply(this, arguments);
+            this.modelName = params.model;
+            this.mode = params.mode;
+            this.options = params.options;
+            this.permissions = params.permissions;
+            this.timeline = params.timeline;
+            this.min_height = params.min_height;
+            this.date_start = params.date_start;
+            this.date_stop = params.date_stop;
+            this.date_delay = params.date_delay;
+            this.colors = params.colors;
+            this.fieldNames = params.fieldNames;
+            this.dependency_arrow = params.dependency_arrow;
+            this.view = params.view;
+            this.modelClass = this.view.model;
+        },
+
+        /**
+         * @override
+         */
+        start: function () {
+            var self = this;
+            var attrs = this.arch.attrs;
+            this.current_window = {
+                start: new moment(),
+                end: new moment().add(24, 'hours'),
+            };
+
+            this.$el.addClass(attrs.class);
+            this.$timeline = this.$el.find(".oe_timeline_widget");
```

Comment:
'this.$()' is a helper function that calls to 'this.$el.find()'
```suggestion
            this.$timeline = this.$(".oe_timeline_widget");
```